exasol_integration_test_docker_environment/test/test_test_env_reuse.py

```diff
@@ -0,0 +1,103 @@
+    '''
+    This test spawns a new test environment and, with parameters:
+    * reuse_database_setup=True,
+    * reuse_database=True,
+    * reuse_test_container=True
+    and verifies if the test data was populated to the docker db.
+    '''
+
+    def env_name(self):
+        return self.__class__.__name__.lower()
+
+    def setUp(self):
+        self._client = docker.from_env()
+        self._docker_repository_name = self.env_name()
+        print("docker_repository_name", self._docker_repository_name)
+        luigi_utils.clean(self._docker_repository_name)
+
+        self.docker_db_version_parameter = test_environment_options.LATEST_DB_VERSION
+        if "EXASOL_VERSION" in os.environ and os.environ["EXASOL_VERSION"] != "default":
+            self.docker_db_version_parameter = os.environ["EXASOL_VERSION"]
+
+    def tearDown(self):
+        luigi_utils.clean(self._docker_repository_name)
+        self._client.close()
+
+    def setup_luigi_config(self):
+        set_docker_repository_config(
+            docker_password=None,
+            docker_repository_name=self._docker_repository_name,
+            docker_username=None,
+            tag_prefix="",
+            kind="target"
+        )
+
+    def run_spawn_test_env(self):
+        result = None
+        self.setup_luigi_config()
+        luigi_utils.set_job_id(SpawnTestEnvironment)
+        task = SpawnTestEnvironment(reuse_database_setup=True,
+                                    reuse_database=True,
+                                    reuse_test_container=True,
+                                    external_exasol_xmlrpc_host="",
+                                    external_exasol_db_host="",
+                                    external_exasol_xmlrpc_port=0,
+                                    external_exasol_db_user="",
+                                    external_exasol_db_password="",
+                                    external_exasol_xmlrpc_user="",
+                                    external_exasol_xmlrpc_password="",
+                                    external_exasol_xmlrpc_cluster_name="",
+                                    external_exasol_bucketfs_write_password="",
+                                    environment_type=EnvironmentType.docker_db,
+                                    environment_name=self.env_name(),
+                                    docker_db_image_version=self.docker_db_version_parameter,
+                                    docker_db_image_name="exasol/docker-db"
+                                    )
+        try:
+            success = luigi.build([task], workers=1, local_scheduler=True, log_level="INFO")
+            if success:
+                result = task
+            else:
+                task.cleanup(False)
+                Exception("Task failed")
+        except Exception as e:
+            task.cleanup(False)
+        return result
+
+    def test_initial_reuse_database_setup_populates_data(self):
```

Comment:
we might need a bit more structure here, it is a bit to much a text wall

Reply:
Done

Reply:
Done